SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorPackageAssembly.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Logging/LogMacros.h"
+#include "Misc/MonitoredProcess.h"
+#include "Widgets/Notifications/SNotificationList.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialGDKEditorPackageAssembly, Log, All);
+
+class SPATIALGDKEDITOR_API FSpatialGDKPackageAssembly : public TSharedFromThis<FSpatialGDKPackageAssembly>
```

Comment:
This lets the object have a weak pointer to itself, so it can be passed in the `BindSP` callbacks, which would check if the object is still valid before executing.